### PR TITLE
[C-2068] Add downloadQueue worker cancellation

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Adds downloadQueue worker cancellation
- Improves how a worker handles reachability changes, cancellation, and download errors in a super elegant and clear way. awesome work @amendelsohn for unlocking this!
- Refactors a lot of the worker code for readability, and moves various queue related sagas together for organization
- Fixed a few bugs with removing items from queue after trying out some crazy edge cases. The new setup is ultra resilient, so stoked!!

### CODE REVIEW

File: packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/downloadCollectionWorker.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      getCollectionOfflineDownloadStatus(collectionId)
+    )
+    if (!trackStatus) return true
+  }
+}
+
+export function* downloadCollectionWorker(collectionId: CollectionId) {
+  yield* put(startDownload({ type: 'collection', id: collectionId }))
+
+  const { downloadCollection, unreachable, cancel } = yield* race({
```
Comment (sliptype) [inline packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/downloadCollectionWorker.ts:44]: 🔥

File: packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/downloadCollectionWorker.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  const [collection] = yield* call([apiClient, apiClient.getPlaylist], {
+    playlistId: collectionId,
+    currentUserId
+  })
+
+  if (
+    !collection ||
+    collection.is_delete ||
+    (collection.is_private && collection.playlist_owner_id !== currentUserId)
+  ) {
```
Comment (sliptype) [inline packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/downloadCollectionWorker.ts:83]: If we are setting the status as error here won't it get re enqueued?

**Review by sliptype — APPROVED** [review]
This is so elegant! Love it

File: packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/downloadCollectionWorker.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/downloadCollectionWorker.ts:83]: its a bit unclear, but "errorDownload" action removes item from queue, so we should be good by next time we request

**Pull request merged**